src/System.ObjectModel/tests/ReadOnlyDictionary/ReadOnlyDictionaryTests.cs

```diff
@@ -217,6 +219,22 @@ public static void DebuggerAttributeTests()
             DebuggerAttributes.ValidateDebuggerDisplayReferences(new ReadOnlyDictionary<int, int>(new Dictionary<int, int>()).Keys);
             DebuggerAttributes.ValidateDebuggerDisplayReferences(new ReadOnlyDictionary<int, int>(new Dictionary<int, int>()).Values);
         }
+
+        [Fact]
+        public static void DebuggerAttribute_NullDictionary_ThrowsArgumentNullException()
+        {
+            TargetInvocationException ex = Assert.Throws<TargetInvocationException>(() =>   DebuggerAttributes.ValidateDebuggerTypeProxyProperties(typeof(ReadOnlyDictionary<int, int>), null));
+            Assert.IsType<ArgumentNullException>(ex.InnerException);
+            Assert.Equal("dictionary", ((ArgumentNullException)ex.InnerException).ParamName);
```

Comment:
IsType returns the cast instance, so you could just store that into a local and then use it, rather than re-accessing ex.InnerException and casting again.